docs/testing-styleguide.MD

```diff
@@ -0,0 +1,280 @@
+
+### Functional Browser Testing
+#### Manual Testing
+Every new or amended component should be fully tested in either Storybook and/or [codesandbox.io](https://codesandbox.io) using functional testing, UI testing and internationalization testing techniques. We use [Jira](https://www.atlassian.com/software/jira) for issue and project tracking. 
+
+##### Browser Support
+Carbon is supported in the following browsers:
+* Chrome
+* Firefox
+* Safari
+* Edge Chromium
+
+Components should be tested in all supported browsers unless the piece of functionality is not specifically related to a UI feature, e.g. adding a data-role prop type.
+
+##### Jira
+For every testable task, a test session needs to be created for each of the supported browsers. Results should be added to each of the conditions in the created test sessions with a pass `(/)` or fail `(x)` mark.
+
+Any issues found during testing should be raised as a "Defect" in Jira unless the ticket is already of a “Bug” type, in which case comments and screenshots/recordings/GIFs can be added to the existing ticket.
+
+If the issue is present on the master branch you should first check if the issue has already been logged in GitHub. If it is present, it should be upvoted with a “+1” comment. If it has not been logged, details of how to do this are contained in the [CONTRIBUTING.MD](https://github.com/Sage/carbon/blob/master/CONTRIBUTING.md) guide. Collaborators internal to Sage should also log a separate Jira ticket for the issue, not blocking this one, as a "Bug" type with the GitHub issue linked. The Jira ticket number should be included in the GitHub issue, e.g. `FE-1234`, but do not include the full Jira URL.
+
+##### Storybook
+Storybook forms our Carbon component library documentation. The component stories are written in `MDX` format containing as many examples as required in set states to showcase our components. Every component should have at least one story (named `default`/`multiple`/`validation` etc.) located in the `Design System` directory. Where required, it should be possible to test all events, e.g. `onClick`, `drag` etc., in these stories. Results of events tests can be found in the `Actions` tab.
+
+Where the stories in the main `Design System` documentation directory do not cover all scenarios required to fully test the component, additional stories (named `basic`/`grouped`/`styles overridden` etc.) must be created in the `Test` directory. For example, scenarios testing special characters, all combinations of size/type/state etc. should have stories in the `Test` directory under the relevant component. Stories for visual testing should also be located here and named `visual`. These stories should be written with the components in set states where possible to avoid the use of knobs although they may be required in some cases, like to allow different text strings to be entered for instance. A developer may need to assist with writing the additional required stories here.
```

Comment:
```suggestion
Where the stories in the main `Design System` documentation directory do not cover all scenarios required to fully test the component, additional stories (named `basic`/`grouped`/`styles overridden` etc.) must be created in the `Test` directory. For example, scenarios testing special characters, all combinations of size/type/state etc. should have stories in the `Test` directory under the relevant component. Stories for visual testing should also be located here in `Test` directory and named `visual`. These stories should be written with the components in set states where possible to avoid the use of knobs although they may be required in some cases, like to allow different text strings to be entered for instance. A developer may need to assist with writing the additional required stories here.
```